### ISSUE TITLE
JS API drops invocations due to potential bug/race condition on GTK platform

### ISSUE THREAD

**Issue by amacf** [r0x0r/pywebview#1006]
### Specification
- pywebview version: 
3.7.2
- operating system:
Ubuntu 22.04
- web renderer: 
GTK webengine

### Description  

When using GTK platform, there appears to be a bug in how it invokes functions via the JS -> Py API mechanism. Namely, if you call two JS_API function in quick succession, one or more of them will never return and will not be invoked on the Python side. 

Here's a minimal example that demonstrates the problem:
```
import webview


html = """
<!DOCTYPE html>
<html>
<head lang="en">
<meta charset="UTF-8">
</head>
<body>


<h1>JS API Example</h1>
<p id='pywebview-status'><i>pywebview</i> is not ready</p>
<div id="response-container"></div>

<script>
    window.addEventListener('pywebviewready', function() {
        var container = document.getElementById('pywebview-status')
        container.innerHTML = '<i>pywebview</i> is ready'
        doTheThings()
    })

    function doTheThings() {
        var container = document.getElementById('response-container')
        pywebview.api.do_thing_1().then(function(response) {
            container.innerHTML += '<p>' + response + '</p>'
        })
        pywebview.api.do_thing_2().then(function(response) {
            container.innerHTML += '<p>' + response + '</p>'
        })
        pywebview.api.do_thing_3().then(function(response) {
            container.innerHTML += '<p>' + response + '</p>'
        })
    }
</script>
</body>
</html>
"""


class Api:
    def do_thing_1(self):
        return "Done thing 1"

    def do_thing_2(self):
        return "Done thing 2"

    def do_thing_3(self):
        return "Done thing 3"


if __name__ == '__main__':
    api = Api()
    window = webview.create_window('API example', html=html, js_api=api)
    webview.start(gui="gtk", debug=True)
```
Executing this with the GTK platform, I consistently only ever get the "Done thing 3" message. It works as-expected on, at least, the Cocoa & QT backends.

Looking at the code, the issue appears to be with how the document title is used as a transport mechanism for invoking the JS API. My assumption of what is happening is that if document.title is set twice by pywebview._bridge.call before GTK's on_title_change is run, then only the last call to set the title will get invoked. 

Assuming that's the issue, a potential fix might be to have to pywebview._bridge.call wait until the python is done reading webview.get_title before it changes the title again. Perhaps the python code calls `webview.run_javascript('document.title=""')` after webview.get_title(), which JS takes as a notification that it can invoke another function. Or something to that effect.
  
### Practicalities

- YES I am willing to work on this issue myself. I can make a PR. Mostly I'd like validation on the problem and suggested fix before proceeding.

- NO I am not prepared to support this issue financially.

**Comment by r0x0r** [comment]
Could you test if this problem still exists in the `4.0` branch? The current js_api implementation for GTK is rather sketchy and buggy, but the new HTTP-server based bridge should be more robust.

**Comment by amacf** [comment]
Thanks, yes, I can confirm the problem is fixed on the 4.0 branch! Is there an expected release date for 4.0?

**Comment by r0x0r** [comment]
In the near future, hopefully within a month.
